test/e2e/extensions/high_availability_test.go

```diff
@@ -111,7 +111,7 @@ func waitForAgonesExtensionsRunning(ctx context.Context) error {
 			return true, err
 		}
 
-		if len(list.Items) != 2 {
+		if len(list.Items) < 2 {
```

Comment:
Correct me if I am wrong, I think extension pods should be more than 1 (https://github.com/googleforgames/agones/blob/main/test/e2e/extensions/high_availability_test.go#L46). Even, we can extend it to more than 2 pods.

Reply:
In our default setup, there should be 2 replica for the extension: https://github.com/googleforgames/agones/blob/main/install/yaml/install.yaml#L17151. This is true for the e2e test cluster. So I am wondering what's the real edge case here? i.e. in what case the replica count goes above 2 and make this test flaky?

Reply:
Interesting question! didn't get the real edge case yet. Also, couldn't replicate it locally with 2 replicas.

Reply:
Curious what made you think it was stuck in this section? 

One recommendation i would also make - is add a bunch of logging here when the test is in this section, so even if this doesn't solve the Flakiness, it should allow us to see what is going on.

For example - how many pods are there on each loop? What are their names? do we need to know anything else about them?

Reply:
Couldn’t get it, if default value for replica of extensions is 2 then how it would have created 4 pods. Even, if we delete 1 pod it will immediately create 1 pod to meet the default replica 2.
Since, I was not able to replicate with 2 replicas, so I manually increased it to 4 so that I can replicate that error locally. And it worked, then I added few logs and found it was exactly failing in this `if condition`(https://github.com/googleforgames/agones/blob/main/test/e2e/extensions/high_availability_test.go#L114).

So, I realize now there is no need to do changes in `if condition` rather I should try to find out why and how it’s making 4 pods.

It has sufficient logs which is printing the pods name and other information. However, I added more logs for my understanding and ran that test multiple times with default replica but couldn’t replicate it and neither I found anything weird.

Reply:
**Assumption:** Either some process are running in parallel which is modifying the replica and due to delay in sync it's giving 4 pods or maybe rolling update would be happening somewhere?

Reply:
So some thoughts:

We could improve the logging here, and wait for it to fail again in CI. I'm particularly interested in seeing the event stream and/or the status values for the Deployment if the replica count goes above 2, and maybe also listing out all the pod names and if they are running/terminating or not.

Handy for logging events:
https://github.com/googleforgames/agones/blob/589f77b4bddac84b6ab225cb62b44aa181b53cbd/test/e2e/framework/framework.go#L829

OH I just had an idea! (and could be good to add to logging too to confirm)

https://github.com/googleforgames/agones/blob/589f77b4bddac84b6ab225cb62b44aa181b53cbd/test/e2e/extensions/high_availability_test.go#L131-L134

What if some of the Pods are stuck Terminating for a while? This function doesn't filter by Terminating pods (`Pod.ObjectMeta.DeletionTimestamp.IsZero()`) -- should we be only looking at Pods that are running?

Reply:
Sure! I have added logs and also filter out the terminating pods. Only giving the running pods.
Could you please review it and let me know if any other changes are required?